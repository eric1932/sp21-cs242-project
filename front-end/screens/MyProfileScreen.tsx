import * as React from 'react';
import {Button, ScrollView, StyleSheet} from 'react-native';
import {View} from '../components/Themed';
import {getToken, removeToken} from "../utils/Storage";
import MyProfileItem from "../components/MyProfileItem";
import {MyProfileProps} from "../types";
import {logoutUser} from "../utils/API";

export default function MyProfileScreen(props: MyProfileProps) {
  let [token, setToken] = React.useState('')

  React.useEffect(() => {
    getToken().then(token => {
      if (token) {
        setToken(token)
      }
    })
  }, [])

  return (
    <View style={styles.container}>
      <ScrollView>
        <MyProfileItem name={"token"} value={token} />
        <Button title={"logout save token"} onPress={async () => {
          // remove token
          await removeToken()
          // go back to login page
          props.navigation.replace('Login')
        }} />
        <Button title={"logout clear token"} onPress={async () => {
          let token = await getToken()
          await removeToken()
          props.navigation.replace('Login')
          // clear token
          if (token) {
            await logoutUser(token, false)
          }
        }} />
        <Button title={"logout everywhere"} onPress={async () => {
          let token = await getToken()
          await removeToken()
          props.navigation.replace('Login')
          // clear token all
          if (token) {
            await logoutUser(token, true)
          }
        }} />
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    // justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
});
