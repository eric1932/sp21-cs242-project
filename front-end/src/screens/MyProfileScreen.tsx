import * as React from 'react';
import {Button, ScrollView, StyleSheet} from 'react-native';
import {View} from '../components/Themed';
import {getToken, removeToken} from "../utils/Storage";
import ListItem from "../components/ListItem";
import {MyProfileProps} from "../types";
import {logoutUser} from "../utils/API";

export default function MyProfileScreen(props: MyProfileProps) {
  const [token, setToken] = React.useState('')

  React.useEffect(() => {
    getToken().then(token => {
      if (token) {
        setToken(token)
      }
    })
  }, [])

  return (
    <View style={styles.container}>
      <ScrollView style={{width: '65%'}}>
        <ListItem name={"token"} value={token} />
        <View style={[styles.listButtonWrapper, {paddingTop: 40}]}>
          <Button title={"logout save token"} onPress={async () => {
            // remove token
            await removeToken()
            // go back to login page
            props.navigation.replace('Login')
          }} />
        </View>
        <View style={styles.listButtonWrapper}>
          <Button title={"logout clear token"} onPress={async () => {
            const token = await getToken()
            await removeToken()
            props.navigation.replace('Login')
            // clear token
            if (token) {
              await logoutUser(token, false)
            }
          }} />
        </View>
        <View style={styles.listButtonWrapper}>
          <Button title={"logout everywhere"} onPress={async () => {
            const token = await getToken()
            await removeToken()
            props.navigation.replace('Login')
            // clear token all
            if (token) {
              await logoutUser(token, true)
            }
          }} />
        </View>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    // justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  listButtonWrapper: {
    width: '50%',
    paddingTop: 15,
    position: 'relative',
    left: '25%'
  }
});
