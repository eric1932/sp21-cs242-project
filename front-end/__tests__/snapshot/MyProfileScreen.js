import * as React from 'react';
import renderer from 'react-test-renderer';
import MyProfileScreen from "../../src/screens/MyProfileScreen";

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
});

it('MyProfileScreen', () => {
  const tree = renderer.create(
    <MyProfileScreen {...createTestProps({})} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
