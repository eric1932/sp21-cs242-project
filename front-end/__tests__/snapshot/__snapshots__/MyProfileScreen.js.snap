// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfileScreen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    style={
      Object {
        "width": "65%",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#eee",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginLeft": "5%",
            "marginRight": "5%",
            "paddingHorizontal": 10,
            "paddingVertical": 10,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              undefined,
            ]
          }
        >
          token
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Array [
              Object {
                "left": "25%",
                "paddingTop": 15,
                "position": "relative",
                "width": "50%",
              },
              Object {
                "paddingTop": 40,
              },
            ],
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              logout save token
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "left": "25%",
              "paddingTop": 15,
              "position": "relative",
              "width": "50%",
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              logout clear token
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Object {
              "left": "25%",
              "paddingTop": 15,
              "position": "relative",
              "width": "50%",
            },
          ]
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={Object {}}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              logout everywhere
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
